src/reducers/poiForm.reducer.js

```diff
@@ -60,6 +62,14 @@ export default function poiForm(state = initialState.poiForm, action) {
           media => media.contentUrl !== action.payload.contentUrl
         )
       }
+    case POI_FORM_CAPTION_MODIFIED:
+      const { captionIndex, captionValue } = action.payload
+      const newMedia = [...state.media]
+      newMedia[captionIndex]['caption'] = captionValue
```

Comment:
you should use `newMedia[captionIndex].caption`. use square brackets when needing to dynamically set it